contrib/hack/migration-check.sh

```diff
@@ -0,0 +1,189 @@
+
+    for className in $(kubectl get serviceinstances --all-namespaces -ojsonpath="{.items[*].status.userSpecifiedClassName}")
+    do
+        type=$(echo "${className}" | cut -d'/' -f1)
+        externalName=$(echo "${className}" | cut -d'/' -f2)
+
+        set +o errexit
+        if [[ "${type}" = "ClusterServiceClass" ]]; then
+            checkIfClassExist "${externalName}" "${clusterServiceClassesNames}"
+            if [[ $? -eq 1 ]]; then
+                echo "${className} not exist in the cluster for the ServiceInstances:"
+                kubectl get serviceinstances -o custom-columns=NAME:.metadata.name,NAMESPACE:.metadata.namespace,STATUS:.status.provisionStatus,CLASS\ NAME:.status.userSpecifiedClassName | grep "${className}"
+                setExitCode
+            fi
+        fi
+        if [[ "${type}" = "ServiceClass" ]]; then
+            checkIfClassExist "${externalName}" "${serviceClassesNames}"
+            if [[ $? -eq 1 ]]; then
+                echo "${className} not exist in the cluster for the ServiceInstances:"
+                kubectl get serviceinstances -o custom-columns=NAME:.metadata.name,NAMESPACE:.metadata.namespace,STATUS:.status.provisionStatus,CLASS\ NAME:.status.userSpecifiedClassName | grep "${className}"
+                setExitCode
+            fi
+        fi
+    done
+    set -o errexit
+}
+
+function setExitCode() {
+    echo
+    EXIT_CODE=1
+}
+# checks if there are some instances with not existing classes
+checkIfClassExistForInstance
+
+#
+# Check if any class/plans were removed from broker's catalog
+#
+CSC=$(kubectl get clusterserviceclasses -ojsonpath="{.items[*].metadata.deletionTimestamp}")
+if [[ -n ${CSC} ]]; then
+    echo "There are being deleted ClusterServiceClasses:"
+    kubectl get clusterserviceclasses -o custom-columns=NAME:.metadata.name,DELETION\ TIME:.metadata.deletionTimestamp
+    setExitCode
+fi
+for status in $(kubectl get clusterserviceclasses -ojsonpath="{.items[*].status.removedFromBrokerCatalog}")
+do
+if [[ -n "${status}" ]] && ${status}; then
+    echo "There are removed from broker's catalog ClusterServiceClasses:"
+    kubectl get clusterserviceclasses -o custom-columns=NAME:.metadata.name,REMOVED:.status.removedFromBrokerCatalog
+    setExitCode
+fi
+done
+
+SC=$(kubectl get serviceclasses --all-namespaces -ojsonpath="{.items[*].metadata.deletionTimestamp}")
+if [[ -n ${SC} ]]; then
+    echo "There are being deleted ServiceClasses:"
+    kubectl get serviceclasses -o custom-columns=NAME:.metadata.name,NAMESPACE:.metadata.namespace,DELETION\ TIME:.metadata.deletionTimestamp
+    setExitCode
+fi
+for status in $(kubectl get serviceclasses --all-namespaces -ojsonpath="{.items[*].status.removedFromBrokerCatalog}")
+do
+if [[ -n "${status}" ]] && ${status}; then
+    echo "There are ServiceClasses removed from broker's catalog:"
+    kubectl get serviceclasses -o custom-columns=NAME:.metadata.name,NAMESPACE:.metadata.namespace,REMOVED:.status.removedFromBrokerCatalog
```

Comment:
add --all-namespaces flag (the same comment for other not cluster-wide resoruces)